test/python/circuit/test_random_circuit.py

```diff
@@ -160,3 +165,241 @@ def test_random_circuit_with_zero_distribution(self):
+                if node.op.name == "measure" or node.op.num_qubits < 2:
+                    continue
+                key = node.sort_key
+                control, target = key.split(",")
+                count_register[(control, target)] += 1
+
+        for occurence in count_register.values():
+            self.assertLessEqual(freq, occurence)
+
+    @ddt.data(*test_cases)
+    @ddt.unpack
+    def test_random_measure(self, inter_graph, seed):
+        """Test random circuit with final measurement."""
+
+        qc = random_circuit_from_graph(
+            interaction_graph=inter_graph, min_2q_gate_per_edge=1, measure=True, seed=seed
+        )
+        self.assertIn("measure", qc.count_ops())
+
+    @ddt.data(*test_cases)
+    @ddt.unpack
+    def test_random_circuit_conditional_reset(self, inter_graph, seed):
+        """Test generating random circuits with conditional and reset."""
+        # Presence of 'reset' in the circuit is probabilistic, at seed 0 reset exists in circuit.
+
+        qc = random_circuit_from_graph(
+            interaction_graph=inter_graph,
+            min_2q_gate_per_edge=2,
+            conditional=True,
+            reset=True,
+            seed=seed,  # Do not change the seed or the args.486
+            insert_1q_oper=True,
+            prob_conditional=0.41,
+            prob_reset=0.50,
+        )
+        self.assertIn("reset", qc.count_ops())
+
+        # Now, checking for conditionals
+
+        conditions = []
+        for instr in qc:
+            cond = getattr(instr.operation, "condition", None)
+            if not cond is None:
+                conditions.append(cond)
+
+        # See if conditionals are present.
+        self.assertNotEqual(conditions, [])
+
+    @ddt.data(*test_cases)
+    @ddt.unpack
+    def test_2q_gates_applied_to_edges_from_interaction_graph(self, inter_graph, seed):
+        """Test 2Q gates are applied to the qubit-pairs given by the interaction graph supplied"""
+
+        qc = random_circuit_from_graph(
+            interaction_graph=inter_graph,
+            min_2q_gate_per_edge=1,
+            measure=True,
+            conditional=True,
+            reset=True,
+            insert_1q_oper=True,
+            seed=seed,
+            prob_conditional=0.21,
+        )
+        dag = circuit_to_dag(qc)
+
+        cp_mp = set()
+        edge_list = set(inter_graph.edge_list())
+
+        for wire in dag.wires:
+            for dag_op_node in dag.nodes_on_wire(wire, only_ops=True):
+                if dag_op_node.op.num_qubits == 2:
+                    control, target = dag_op_node.qargs
+                    control_idx = control._index
+                    target_idx = target._index
+                    cp_mp.update({(control_idx, target_idx)})
+
+        # make sure every qubit-pair from the circuit actually present in the edge_list
+        for cp in cp_mp:
+            self.assertTrue(cp in edge_list)
+
+    def test_2q_gates_excluded_edges_with_zero_weight(self):
```

Comment:
I am doing code duplication because I want to take a test `cp_map` with zero weight for this test only.